### PR TITLE
Refactor and clean up twitch.cpp

### PR DESCRIPTION
**Pull request by IntelOrca**
To conform with new CPP code conventions. Also tidy up VS filters.

### CODE REVIEW

File: src/network/twitch.cpp [hunk 1]
```diff
@@ -1,449 +1,577 @@
...
+        }
+    };
+
+    /**
+     * The time between HTTP requests.
+     * TODO Ideally, the chat message pulse should be more frequent than the followers / chat members so that news messages etc.
+     * have a lower latency.
+     */
+    constexpr uint32 PulseTime = 10 * 1000;
+    constexpr char * TwitchExtendedBaseUrl = "http://openrct.ursalabs.co/api/1/";
```
Comment (janisozaur) [inline src/network/twitch.cpp]:
```cpp
/home/travis/build/janisozaur/OpenRCT2/src/network/twitch.cpp:90:46: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]

     constexpr char * TwitchExtendedBaseUrl = "http://openrct.ursalabs.co/api/1/";

                                              ^
```

File: src/core/Diagnostics.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
```
Comment (janisozaur) [inline src/core/Diagnostics.cpp]:
```cpp
/home/travis/build/janisozaur/OpenRCT2/src/core/Diagnostics.cpp:1:9: warning: #pragma once in main file [enabled by default]

 #pragma once

         ^
```

File: openrct2.vcxproj.filters [hunk 3]
```diff
@@ -890,13 +884,23 @@
...
+    </ClInclude>
+    <ClInclude Include="src\core\Json.hpp">
+      <Filter>Source\Core</Filter>
+    </ClInclude>
+    <ClInclude Include="src\core\List.hpp">
+      <Filter>Source\Core</Filter>
+    </ClInclude>
   </ItemGroup>
-</Project>
+</Project>
```
Comment (janisozaur) [inline openrct2.vcxproj.filters]: The fight will never end :P

Comment (IntelOrca) [inline openrct2.vcxproj.filters]: VS will keep changing it back, just let it be.

**Comment by janisozaur** [comment]
Requires rebase

**Pull request merged**